### PR TITLE
Additional info about Snowflake provider breaking amazon one

### PR DESCRIPTION
**Pull request by potiuk**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
Turns out snowflake provider also breaks Amazon provider :(

**Comment by potiuk** [comment]
I've added extra info to both Amazon and Snowflake providers.

File: airflow/providers/amazon/ADDITIONAL_INFO.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ software distributed under the License is distributed on an
+ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied.  See the License for the
+ specific language governing permissions and limitations
+ under the License.
+ -->
+
+## Additional limitations
+
+The amazon provider might not work correctly with when it is installed together with the
```
Comment (turbaszek) [inline airflow/providers/amazon/ADDITIONAL_INFO.md]:
```suggestion
The amazon provider might not work correctly when it is installed together with the
```

File: airflow/providers/amazon/ADDITIONAL_INFO.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+Snowflake provider. It's because of Snowflake monkeypatching the urllib3
+library as described in [this issue](https://github.com/snowflakedb/snowflake-connector-python/issues/324)
```
Comment (turbaszek) [inline airflow/providers/amazon/ADDITIONAL_INFO.md]:
```suggestion
library as described in [this issue](https://github.com/snowflakedb/snowflake-connector-python/issues/324),
```

File: airflow/providers/snowflake/ADDITIONAL_INFO.md [hunk 3]
```diff
@@ -17,10 +17,16 @@
...
+## Known incompatibilities with other providers
 
-If you are upgrading from 2020.10.5 note the following changes in import paths
+The Snowflake provider when installed might break other providers. As of 14 December 2020 we know that Presto with
+Kerberos + SSL and Amazon providers are broken.
 
-| Old path                                                        | New path                                                    |
-| --------------------------------------------------------------- | ----------------------------------------------------------- |
-| airflow.providers.amazon.aws.hooks.aws_dynamodb.AwsDynamoDBHook | airflow.providers.amazon.aws.hooks.dynamodb.AwsDynamoDBHook |
+It's because of Snowflake monkeypatching the urllib3library as described in
```
Comment (kaxil) [inline airflow/providers/snowflake/ADDITIONAL_INFO.md]:
```suggestion
It's because of Snowflake monkeypatching the `urllib3` library as described in
```

**Comment by potiuk** [comment]
Committed all of those

**Review by kaxil — APPROVED** [review]

**Comment by potiuk** [comment]
It looks like I would have to add this information to almost all the providers, so I am holding of with this one - we discussed with @ashb that we will immediately yank the snowflake provider to mark it as "bad" release and we are informing Snowflake to fix their client. They seem to realize that already that this is VERY bad thing they'e done.

**Pull request closed without merging**